src/LightGBM.jl

```diff
@@ -39,7 +40,16 @@ end
 
 
 function __init__()
-    LGBM_library[] = Libdl.dlopen(find_library("lib_lightgbm", [@__DIR__]))
+    if Sys.isapple()
+        # for macOS users alternative LightGBM_jll - binaries compiled with Binary Builder
+        # particularly to address the issues raised for M1/aarch64-apple-darwin
+        @eval using LightGBM_jll
+        LGBM_library[] = Libdl.dlopen(find_library("lib_lightgbm", [@__DIR__]))
```

Comment:
```
LGBM_library[] = Libdl.dlopen(find_library("lib_lightgbm", [@__DIR__]))
```
This appears to work when testing it on aarch64. I have a `LIGHTGBM_EXAMPLES_PATH` failure, but apart from that the `MLJ interface tests` test set passes.

Reply:
This is because some tests are directly using Microsoft lightgbm examples so to run these tests you need to export the LIGHTGBM_EXAMPLES_PATH to the location where these examples are downloaded like it's mentioned [here](https://github.com/IQVIA-ML/LightGBM.jl#tests).

Reply:
Thank you @kainkad. To verify, I just ran this `binary-builder` branch on an aarch64 laptop with `MLJBase = "0.21"` in `test/Project.toml` and all tests pass.